parser-library/buffer.cpp

```diff
@@ -52,6 +52,22 @@ struct buffer_detail {
 #endif
 };
 
+#ifndef WIN32
+uint16_t _byteswap_ushort(uint16_t val) {
+  return ((val >> 8) & 0xFF) | ((val << 8) & 0xFF00);
+}
+
+uint32_t _byteswap_ulong(uint32_t val) {
+  return (_byteswap_ushort(val >> 16) & 0xFFFF) |
+          ((((uint32_t) _byteswap_ushort(val)) << 16) & 0xFFFF0000);
+}
+
+uint64_t _byteswap_uint64(uint64_t val) {
+  return (_byteswap_ulong(val >> 32) & 0xFFFFFFFF) |
+          ((((uint64_t) _byteswap_ulong(val)) << 32) & 0xFFFFFFFF00000000);
```

Comment:
Below you use `reinterpret_cast`, I'd like to see you use `static_cast` here instead of a C-style cast.

For the immediate constants, `0xFF...`, append on the `U`, `UL` and `ULL` suffixes so that the results aren't converted to `int`s.

it's possible that the shift result will also be silently converted to an `int`.